src/Compilers/CSharp/Portable/Binder/ScriptInitializerBinder.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed class ScriptInitializerBinder : LocalScopeBinder
+    {
+        private readonly ImmutableArray<LabelSymbol> _labels;
+
+        internal ScriptInitializerBinder(SynthesizedInteractiveInitializerMethod scriptInitializer, CompilationUnitSyntax syntax, Binder next) : base(next)
+        {
+            _labels = GetLabels(scriptInitializer, syntax);
+        }
+
+        protected override ImmutableArray<LabelSymbol> BuildLabels()
+        {
+            return _labels;
```

Comment:
It looks like this was intended to be lazy.  Do we have a reason for not following the pattern?